lib/utils/isStandardSyntaxRule.js

```diff
@@ -37,6 +37,11 @@ module.exports = function (rule/*: Object*/)/*: boolean*/ {
     return false
   }
 
+  // Less guards
+  if ((/when.*?\(.*?\)/).test(selector)) {
```

Comment:
How about `/when\s+(not\s+)*\(/`? I think it will just take writing tests for what should and should not work and fiddling with the regexp until that's what happens.